src/main/py/hl7_listener/main.py

```diff
@@ -9,31 +9,48 @@
 - NATS JetStream server is running and configured with expected Subject.
 """
 import asyncio
-
+import certifi
 import os
 from hl7_listener import logger_util, logging_codes
 import hl7
+import ssl
 from hl7.mllp import start_hl7_server
 from nats.aio.client import Client as NATS
 from nats.aio.errors import ErrNoServers
+from nats.js import JetStreamContext
+from typing import Optional
+
 
 logger = logger_util.get_logger(__name__)
 
 # HL7 is the Stream and ENCRYPTED_BATCHES is the Consumer.
 _subject = os.getenv("NATS_OUTGOING_SUBJECT", default="HL7.MESSAGES")
 # NATS Jetstream connection info
 _nats_server_url = os.getenv("NATS_SERVER_URL")
+_nats_allow_reconnect = bool(os.getenv("NATS_ALLOW_RECONNECT", default=True))
```

Comment:
As we iterate towards a more upstream integration, it would be great to consider Pydantic BaseSettings models. That would encapsulate the settings in a single model and provide a bit of type safety we're achieving with explicit conversions.